File: WebStoreHubAPI/Models/ProductModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using static PdfSharp.Capabilities.Features;

namespace WebStoreHubAPI.Models
{
    public class ProductModel
    {
        [Key]
        public int ProductId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public decimal? DiscountedPrice { get; set; } 
        public int Stock { get; set; }

        public int ProductTypeId { get; set; }
        public int BrandId { get; set; }

        [ForeignKey("ProductTypeId")]
        public ProductTypeModel ProductType { get; set; }

        [ForeignKey("BrandId")]
        public BrandModel Brand { get; set; }
        public List<ImgbbModel> Images { get; set; }

        public List<DiscountModel> Discounts { get; set; }
    }
}

File: angular-frontend/src/app/models/product-model.ts
export interface Product {
  productId: number;
  name: string;
  description: string;
  price: number;
  discountedPrice?: number;
  discountPercentage?: number;
  brand: { brandId: number; brandName: string };
  productType: { productTypeId: number; typeName: string };
  stock: number;
  images: { imageUrl: string; mainPicture: number }[];
  imageUrl?: string;
}
